components/agenda/src/init.spec.js

```diff
@@ -1,202 +1,337 @@
-      // First tick is 12 AM regardless of auto_time_box;
-      cy.get(".ticks").get(".tick:eq(0)").should("contain", "12 AM");
-      // it gets its height offset into the negative if we're timeboxed. Test that.
-      cy.get(".ticks").get(".tick:eq(0)").should("have.css", "top", "0px");
     });
+
+    cy.get("@agenda").contains("12 AM").should("be.visible");
+    cy.get("@agenda").contains("1 AM").should("be.visible");
+    cy.get("@agenda").contains("2 AM").should("be.visible");
+    cy.get("@agenda").contains("3 AM").should("be.visible");
+    cy.get("@agenda").contains("4 AM").should("be.visible");
+    cy.get("@agenda").contains("5 AM").should("be.visible");
+    cy.get("@agenda").contains("6 AM").should("be.visible");
+    cy.get("@agenda").contains("7 AM").should("be.visible");
+    cy.get("@agenda").contains("8 AM").should("be.visible");
+    cy.get("@agenda").contains("9 AM").should("be.visible");
+    cy.get("@agenda").contains("10 AM").should("be.visible");
+    cy.get("@agenda").contains("11 AM").should("be.visible");
+    cy.get("@agenda").contains("12 PM").should("be.visible");
+    cy.get("@agenda").contains("1 PM").should("be.visible");
+    cy.get("@agenda").contains("2 PM").should("be.visible");
+    cy.get("@agenda").contains("3 PM").should("be.visible");
+    cy.get("@agenda").contains("4 PM").should("be.visible");
+    cy.get("@agenda").contains("5 PM").should("be.visible");
+    cy.get("@agenda").contains("6 PM").should("be.visible");
+    cy.get("@agenda").contains("7 PM").should("be.visible");
+    cy.get("@agenda").contains("8 PM").should("be.visible");
+    cy.get("@agenda").contains("9 PM").should("be.visible");
+    cy.get("@agenda").contains("10 PM").should("be.visible");
+    cy.get("@agenda").contains("11 PM").should("be.visible");
   });
-  it("correctly scales view to start and end when events do", () => {
-    cy.visit("/components/agenda/src/index.html");
-    cy.get("nylas-agenda").then((element) => {
+
+  it("shows compacted timeline from first event to last event when auto_time_box is true", () => {
+    cy.get("@agenda").then((element) => {
       const agenda = element[0];
-      agenda.events = AGENDA_EVENTS;
       agenda.auto_time_box = true;
-      // first event starts at 10:00am; or 600 minutes into a 1440 minute day.
-      // last event ends at 2:00pm, or 840 minutes into a 1440 minute day
-      // so we're zooming to between 10a-2p, or 600-840, showing 240/1440 minutes.
-      // Our active view therefore shows 4 hours, and are concealing 10 hours before;
-      // 10/4 = 2.5.
-      // We thus expect the "concealed before" viewport to be 250% of our active view,
-      // in otherwords, a top of -250% (or, against our viewport, -1760px)
-      // (viewport height: 800px; .ticks container: 704px. 704 x 2.5 = 1760)
-      cy.get(".ticks").get(".tick:eq(0)").should("have.css", "top", "-1760px");
     });
+
+    cy.get("@agenda").contains("9 AM").should("not.be.visible");
```

Comment:
Can we just create a for loop for these assertions instead?

Reply:
Same thing for the other cases where we're testing the visibility of ticks

Reply:
I find for loops or most kinds of efficiencies in tests make them less obvious and a little harder to change, so I opted for the obvious approach.

Tradeoff is more lines of code, but they would perform the same. Not too worried about lines of code for tests as long as its easy to update.

Reply:
I'd prefer us to treat tests like any other piece of production code. Duplicated code is hard to manage over time, and I don't think a couple of loops in here would make it too complicated.

The boilerplate for these tests could just look something like
```
const expectedVisibleTimes = ["1 AM", "2 AM", "3 AM", "4 AM", "5 AM"];
const expectedHiddenTimes = ["6 AM", "7 AM", "8 AM", "9 AM"];

for (const time of expectedVisibleTimes) {
    cy.get("@agenda").contains(time).should("be.visible")
}

for (const time of expectedHiddenTimes) {
    cy.get("@agenda").contains(time).should("not.be.visible")
}
```